src/main/resources/templates/db_scripts/publication-period/migratePublicationPeriod.sql

```diff
@@ -0,0 +1,127 @@
+CREATE OR REPLACE FUNCTION ${myuniversity}_${mymodule}.migrate_publication_period(
+    jsonb_data jsonb
+) RETURNS jsonb AS $$
+DECLARE
+    pub_period jsonb;
+    start_date text;
+    end_date text;
+    date_type_id text;
+    dates jsonb := '{}'::jsonb;
+BEGIN
+    pub_period := jsonb_data -> 'publicationPeriod';
+    start_date := pub_period ->> 'start';
+    end_date := pub_period ->> 'end';
+
+    -- Determine Date type
+    IF (start_date IS NOT NULL AND end_date IS NOT NULL) THEN
+        date_type_id := '8fa6d067-41ff-4362-96a0-96b16ddce267';
+    ELSIF (start_date IS NOT NULL OR end_date IS NOT NULL) THEN
+        date_type_id := '24a506e8-2a92-4ecc-bd09-ff849321fd5a';
+    ELSE
+        RETURN jsonb_data;
```

Comment:
`jsonb_data := jsonb_data - 'publicationPeriod';`
should be added before RETURN

Reply:
done